assembl/static2/js/app/components/debate/common/headerActions.jsx

```diff
@@ -0,0 +1,46 @@
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import { Translate } from 'react-redux-i18n';
+import { openShareModal } from '../../../utils/utilityManager';
+
+type Props = {
+  routerParams: Map<string>,
+  ideaId: string,
+  useSocialMedia: boolean
+};
+
+const headerActions = ({ routerParams, ideaId, useSocialMedia }: Props) => {
+  const modalTitle = <Translate value="debate.shareThematic" />;
+  return (
+    <div className="header-actions-container">
+      <div
+        className="share-button action-button"
+        onClick={() =>
+          openShareModal({
+            title: modalTitle,
+            routerParams: routerParams,
+            elementId: ideaId,
+            social: useSocialMedia
+          })
+        }
+      >
+        <div className="share-icon-container white">
+          <span className="assembl-icon-share" />
+        </div>
+        <div className="action-button-label">
+          <Translate value="debate.share" />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const mapStateToProps = ({ debate }) => {
```

Comment:
since you already have `debate` in the `Header` component, you should probably just pass `useSocialMedia` to `HeaderActions` and remove the redux connect